dor/providers/parsers.py

```diff
@@ -59,45 +65,51 @@ def get_data_files(self) -> list[FileMetadata]:
             for elem in self.tree.findall(".//METS:file")
         ]
 
-    def get_md_file_metadatum(self, elem):
+    def get_md_file_metadatum(self, elem: ElementAdapter):
         id_ = elem.get("ID")
         use = elem.get("USE")
-        locref = elem.find("METS:mdRef").get_optional("LOCREF")
-        mdtype = elem.find("METS:mdRef").get_optional("MDTYPE")
-        mimetype = elem.find("METS:mdRef").get_optional("MIMETYPE")
+        md_red_element = elem.find("METS:mdRef")
+        
+        if md_red_element:
+            locref = self.file_provider.apply_relative_path(self.data_path, md_red_element.get_optional("LOCREF"))
+            mdtype = md_red_element.get_optional("MDTYPE")
+            mimetype = md_red_element.get_optional("MIMETYPE", None)
```

Comment:
`get_optional` returns `None` as a default if you don't provide a second argument. Doesn't hurt to be explicit, but doing it both ways here might confuse some.